tasmota/xdrv_16_tuyamcu.ino

```diff
@@ -808,7 +808,7 @@ void TuyaSetTime(void) {
   uint16_t payload_len = 8;
   uint8_t payload_buffer[8];
   payload_buffer[0] = 0x01;
-  payload_buffer[1] = (uint8_t)RtcTime.year;
+  payload_buffer[1] = RtcTime.year - 2000;
```

Comment:
Well, this assumes that RTC's date is behind year 2000. A better fix would be year%100.

Reply:
Agree. Done. Thx.